src/filters/truncate.js

```diff
@@ -3,7 +3,7 @@
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
     return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+      if (typeof str !== 'string' || str.length <= charLimit) {
```

Comment:
Gotcha, making an update now for a quick check on newlines prior to returning.

Regarding the second part about adding/subtracting characters... going to prefix it with a question. 
- Are we aiming to do whole word truncates, where it properly rounds a word to completion, or do we just want to cut the string? 

If we're going for a ```charLimit truncate``` as an end result we're good to go. If we want to end on a word boundary when ```useWordBoundary``` is active then we may need to modify the use of [```charLimit - 10``` on line 24](https://github.com/openshift/origin-web-common/blob/master/src/filters/truncate.js#L24)